src/imitation/algorithms/bc_torch.py

```diff
@@ -0,0 +1,275 @@
+"""Behavioural Cloning (BC).
+
+Trains policy by applying supervised learning to a fixed dataset of (observation,
+action) pairs generated by some expert demonstrator.
+"""
+
+import collections
+from typing import Any, Callable, Dict, List, Mapping, Optional, Tuple, Type, Union
+
+import gym
+import numpy as np
+import torch as th
+from stable_baselines import logger
+from stable_baselines3.common.policies import ActorCriticPolicy
+from stable_baselines3.common.utils import get_schedule_fn, update_learning_rate
+from tqdm.autonotebook import trange
+
+from imitation.data import dataset, types
+from imitation.policies3.base3 import FeedForward32Policy
+
+
+class BC:
+    def __init__(
+        self,
+        observation_space: gym.Space,
+        action_space: gym.Space,
+        *,
+        policy_class: Type[ActorCriticPolicy] = FeedForward32Policy,
+        policy_kwargs: Optional[Mapping[str, Any]] = None,
+        expert_data: Union[
+            types.Transitions, dataset.Dataset[Dict[str, np.ndarray]], None
+        ] = None,
+        batch_size: int = 32,
+        optimizer_cls: Type[th.optim.Optimizer] = th.optim.Adam,
+        optimizer_kwargs: Optional[Dict[str, Any]] = None,
+        ent_weight: float = 1e-3,
+        l2_weight: float = 0.0,
+    ):
+        """Behavioral cloning (BC).
+
+        Recovers a policy via supervised learning on a Dataset of observation-action
+        pairs.
+
+        This class borrows code from Stable Baselines 3
```

Comment:
Originally, I copied the functions `_update_current_progress_remaining` and `_update_learning_rate` from `stable_baselines3.common.base_class.BaseAlgorithm`. Now, I've pushed a new commit that just calls these functions from `stable_baselines3.common.base_class.BaseAlgorithm` instead of defining them again in bc.